Cython/Compiler/Symtab.py

```diff
@@ -2374,7 +2387,6 @@ def declare_var(self, name, type, pos,
             # xxx: is_pyglobal changes behaviour in so many places that I keep it in for now.
             # is_member should be enough later on
             entry.is_pyglobal = 1
-            self.namespace_cname = "(PyObject *)%s" % self.parent_type.typeptr_cname
```

Comment:
This is moved because `self.parent_type` isn't yet defined in the constructor (and I now call `declare_var` in the constructor). I therefore calculate it lazily instead.

Reply:
Don't we know the parent type at the time where we call the constructor?
Lazily calculating something generally adds a complexity that needs to be well motivated.

Reply:
Possibly. It looks like we should be able to in the most common case anyway. I'll have a go and see if it can be made to work easily